dotnet/src/SemanticKernel.Abstractions/Contents/ChatMessageContent.cs

```diff
@@ -24,6 +24,7 @@ public class ChatMessageContent : KernelContent
     /// A convenience property to get or set the text of the first item in the <see cref="Items" /> collection of <see cref="TextContent"/> type.
     /// </summary>
     [EditorBrowsable(EditorBrowsableState.Never)]
+    [JsonIgnore]
```

Comment:
I wonder what would happen here if someone:  
1. Has chat history serialized and stored.  
2. Updates their code to the SK with the change.  
3. Deserializes the chat history.  
   
Would the property be assigned since it exists in the serialized JSON, despite the property being marked as ignored? Or would there be data loss?

Reply:
There is no data loss.  If you are referring to the `Content` property, this is a derived/computed property and doesn't need to particiapte in the serialization contract.  This `Content` property will be ignored, but the history will not have any data-loss since the state of this derived/computed property is informed by the items collection.

It is common for properties in JSON not have a reciever property in the corresponding object definition.

Reply:
It began being computed quite recently - just one month ago. Before that, it was merely a getter and setter. My concern is that there might be a serialized version of the chat history out there, however, the chances may be very low, with a value for the Content but without the value in the Items collection.

Reply:
Can we ignore for serialisation but use the value for deserialisation?

Reply:
I'm address the current implementation along with other optimizations.  I commonly create console projects to demonstrate to myself certain behaviors.  In this case, this is a non-breaking & non-data loss change.  (I also agree that this is a very low likelihood at this point as well)

Reply:
Previously chaninging the nature of the `Content` property sematatics is an unfortunate evolution of the data-contract that might be best addressed with a release note given the low likelyhood that this case even exists in the wild.

Reply:
> Can we ignore for serialisation but use the value for deserialisation?

I'll review what possible.

Reply:
FYI - Follow-up / validation on my deserialization behavior description: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/missing-members

Reply:
I don't believe so.  I'm not seeing an ingore on write-only option...perhaps with a customer serializer, that that pushes requirements to the caller.

https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/ignore-properties
https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonignorecondition

Reply:
Sergey has already identified this is a likely non-issue.  I think we just need blessing on release note.

Reply:
Since both you and Sergey have identified this as a rare / non-issue, I'm ok with proceeding.

Reply:
Let's def call out in release note.  Is there a way to track next release notes?

Reply:
@markwallace-microsoft, are you able to provide guidance on how to add items to the release notes? Is it possible to also add some sample code demonstrating how to massage the previous serialization if someone is using the old version and has them saved in something like COSMOS DB?

Reply:
I'm happy to gin up a translatin sample.

Reply:
@crickman The information in the PR description and details are what get's used in the release notes but we can also add additional content. So let's start with making sure the PR details highlights the potential issue.